packages/@aws-cdk/aws-iam/lib/oidc-provider.ts

```diff
@@ -0,0 +1,154 @@
+   * provider. For example, assume that the OIDC provider is server.example.com
+   * and the provider stores its keys at
+   * https://keys.server.example.com/openid-connect. In that case, the
+   * thumbprint string would be the hex-encoded SHA-1 hash value of the
+   * certificate used by https://keys.server.example.com.
+   *
+   * @default - If no thumbprints are specified (an empty array or `undefined`),
+   * the thumbprint of the root certificate authority will be obtained from the
+   * provider's server as described in https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+   */
+  readonly thumbprints?: string[];
+}
+
+/**
+ * IAM OIDC identity providers are entities in IAM that describe an external
+ * identity provider (IdP) service that supports the OpenID Connect (OIDC)
+ * standard, such as Google or Salesforce. You use an IAM OIDC identity provider
+ * when you want to establish trust between an OIDC-compatible IdP and your AWS
+ * account. This is useful when creating a mobile app or web application that
+ * requires access to AWS resources, but you don't want to create custom sign-in
+ * code or manage your own user identities.
+ *
+ * @see http://openid.net/connect
+ * @see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html
+ *
+ * @experimental
+ */
+export class OpenIdConnectProvider extends Construct implements IOpenIdConnectProvider {
+  /**
+   * Imports an Open ID connect provider from an ARN.
+   * @param scope The definition scope
+   * @param id ID of the construct
+   * @param openIdConnectProviderArn the ARN to import
+   */
+  public static fromOpenIdConnectProviderArn(scope: Construct, id: string, openIdConnectProviderArn: string): IOpenIdConnectProvider {
+    class Import extends Resource implements IOpenIdConnectProvider {
+      public readonly openIdConnectProviderArn = openIdConnectProviderArn;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The Amazon Resource Name (ARN) of the IAM OpenID Connect provider.
+   */
+  public readonly openIdConnectProviderArn: string;
+
+  /**
+   * Defines an OpenID Connect provider.
+   * @param scope The definition scope
+   * @param id Construct ID
+   * @param props Initialization properties
+   */
+  public constructor(scope: Construct, id: string, props: OpenIdConnectProviderProps) {
+    super(scope, id);
+
+    const resource = new CustomResource(this, 'Resource', {
+      resourceType: RESOURCE_TYPE,
+      serviceToken: this.getOrCreateProvider(),
+      properties: {
+        ClientIDList: props.clientIds,
+        ThumbprintList: props.thumbprints,
+        Url: props.url,
+      },
+    });
+
+    this.openIdConnectProviderArn = Token.asString(resource.ref);
+  }
+
+  public get stack() { return Stack.of(this); }
+
+  private getOrCreateProvider() {
+    return CustomResourceProvider.getOrCreate(this, RESOURCE_TYPE, {
```

Comment:
Timeout?

Reply:
The default timeout for CR providers is 15min.